plugins/test/mall_motor_vehicle.js

```diff
@@ -0,0 +1,126 @@
+                            <nd ref="-13" />\
+                            <nd ref="-10" />\
+                            <tag k="F_CODE" v="AG030"/>\
+                        </way>\
+                    </osm>';
+
+        var osm_xml = server.handleInputs({
+            osm: data,
+            method: 'POST',
+            translation: 'TDSv61',
+            path: '/translateFrom'
+        });
+
+        // console.log(osm_xml);
+
+        var xml = parser.parseFromString(osm_xml);
+        var gj = osmtogeojson(xml);
+
+        assert.equal(xml.getElementsByTagName("osm")[0].getAttribute("schema"), "OSM");
+
+        var tags = gj.features[0].properties;
+        assert.equal(tags["shop"], "mall");
+
+        var tds_xml = server.handleInputs({
+            osm: osm_xml,
+            method: 'POST',
+            translation: 'TDSv61',
+            path: '/translateTo'
+        });
+
+        // console.log(tds_xml);
+        
+        xml = parser.parseFromString(tds_xml);
+        gj = osmtogeojson(xml);
+
+        assert.equal(xml.getElementsByTagName("osm")[0].getAttribute("schema"), "TDSv61");
+
+        var tags = gj.features[0].properties;
+        assert.equal(tags["F_CODE"], "AG030");
+
+    });
+
+
+    it('should translate Motor Vehicle Station (AG030) from tdsv61 -> osm -> tdsv61', function() {
+
+        var data = '<osm version="0.6" upload="true" generator="hootenanny">\
+                        <node id="-10" action="modify" visible="true" lat="0.68307256979" lon="18.45073925651" />\
+                        <node id="-11" action="modify" visible="true" lat="0.68341620728" lon="18.45091527847" />\
+                        <node id="-12" action="modify" visible="true" lat="0.68306209303" lon="18.45157116983" />\
+                        <node id="-13" action="modify" visible="true" lat="0.68270797876" lon="18.45141400736" />\
+                        <way id="-19" action="modify" visible="true">\
+                            <nd ref="-10" />\
+                            <nd ref="-11" />\
+                            <nd ref="-12" />\
+                            <nd ref="-13" />\
+                            <nd ref="-10" />\
+                            <tag k="F_CODE" v="AQ170"/>\
+                            <tag k="FFN" v="470"/>\
+                            <tag k="FFN2" v="382"/>\
+                        </way>\
+                    </osm>';
+
+        var osm_xml = server.handleInputs({
+            osm: data,
+            method: 'POST',
+            translation: 'TDSv61',
+            path: '/translateFrom'
+        });
+
+        // console.log(osm_xml);
+
+        var xml = parser.parseFromString(osm_xml);
+        var gj = osmtogeojson(xml);
+
+        assert.equal(xml.getElementsByTagName("osm")[0].getAttribute("schema"), "OSM");
+
+        var tags = gj.features[0].properties;
+        assert.equal(tags["amenity"], "fuel");
+        assert.equal(tags["toilets:2"], "yes");
```

Comment:
But you get a bit of weirdness with the osm tag `toilets:2`.  If the osm tag is just `toilets` then the extra feature function gets mapped to the MEMO field and is effectively hidden in the UI.  I'm not sure which is better, though both are symmetrical with the noted pros/cons.

Reply:
Ahhhh. Time for another ticket. I think the FFN, FFN2, FFN3 code needs to collapse the attributes.
E.g. 
`amenity=fuel` goes to FFN.  
`toilets=yes` goes to FFN2 since FFN is already assigned.

Going to OSM from TDS, the translation should check if the key is set and if it is, then add `:2` or `:3` etc.